tfhe/src/core_crypto/fft_impl/fft128_u128/math/fft/mod.rs

```diff
@@ -206,10 +206,11 @@ pub fn wrapping_add_avx2(
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 #[inline(always)]
 pub fn wrapping_neg_avx2(simd: V3, (lo, hi): (u64x4, u64x4)) -> (u64x4, u64x4) {
-    let diff_lo = pulp::cast(simd.wrapping_sub_u64x4(simd.splat_u64x4(0), lo));
-    let overflow = pulp::cast(simd.cmp_lt_u64x4(simd.splat_u64x4(0), lo));
-    let diff_hi = simd.wrapping_sub_u64x4(overflow, hi);
-    (diff_lo, diff_hi)
+    wrapping_add_avx2(
+        simd,
+        (simd.splat_u64x4(1), simd.splat_u64x4(0)),
+        (simd.not_u64x4(lo), simd.not_u64x4(hi)),
+    )
```

Comment:
the old formula was wrong because i was using a mask as if it had a 0 or 1 value, when it actually has a 0 or u64::MAX value

the new formula uses the fact that in two's complement, we have !x + x + 1 = 0 mod 2^N
which means that -x = !x + 1 mod 2^N

Reply:
ok understood, thanks a lot !